### PR TITLE
Add tags tracking [MAILPOET-4583]

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

This pull request extends tracking.

## Code review notes

_N/A_

## QA notes

New properties are visible in Mixpanel.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4583]

## After-merge notes

_N/A_

[MAILPOET-4583]: https://mailpoet.atlassian.net/browse/MAILPOET-4583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/analytics.js [hunk 1]
```diff
@@ -12,3 +12,51 @@ export function getTrackingData() {
...
+  if (filterType === 'woocommerce' && action === 'purchasedCategory')
+    return 'purchased in category';
+  if (filterType === 'woocommerce' && action === 'purchasedProduct')
+    return 'purchased product';
+  if (filterType === 'userRole' && action === 'subscribedDate')
+    return 'subscribed date';
+  if (filterType === 'woocommerce' && action === 'totalSpent')
+    return 'total spent';
+  if (filterType === 'woocommerce' && action === 'totalSpent')
+    return 'total spent';
```
Comment (triple0t) [inline mailpoet/assets/js/src/analytics.js]: This part is Repeated.

```
if (filterType === 'woocommerce' && action === 'totalSpent')
    return 'total spent';
  if (filterType === 'woocommerce' && action === 'totalSpent')
    return 'total spent';
```

File: mailpoet/assets/js/src/analytics.js [hunk 2]
```diff
@@ -12,3 +12,51 @@ export function getTrackingData() {
...
+    return 'subscribed to list';
+  if (filterType === 'email' && action === 'opened') return 'opened';
+  if (filterType === 'email' && action === 'machineOpened')
+    return 'machine-opened';
+  if (filterType === 'woocommerceMembership' && action === 'isMemberOf')
+    return 'is active member of';
+  if (
+    filterType === 'woocommerceSubscription' &&
+    action === 'hasActiveSubscription'
+  )
```
Comment (triple0t) [inline mailpoet/assets/js/src/analytics.js]: I would like to believe there's a better way to handle this block of code, but my brain is too tired to think right now.

I do think it makes readability a little challenging and makes updates tricky.

A quick suggestion I may offer:

Try to align and order by the same type of `filterType`
e.g
if filterType === 'email'

Show all filterType === 'email' before showing filterType === 'userRole' or filterType === 'woocommerce'

In this case, we have filterType === 'email', filterType === 'woocommerce' then filterType === 'email' again.

**Review by triple0t — DISMISSED** [review]
Looks good, @lysyjan. Well done.

I have some comments, but I don't think they are major.

You may choose to go ahead and push to QA, or you may choose to update the PR.

I'm assigning the PR back to you to make your choice.

File: mailpoet/assets/js/src/analytics.js (hunk 2 above)
Comment (samnajian) [reply mailpoet/assets/js/src/analytics.js]: Sorry for jumping in, in these cases I really like the pattern used for [creating reducers](https://github.com/kolodny/redux-create-reducer#readme) which is used for [NgRX reducers](https://ngrx.io/guide/store/reducers) as well, which is basically only a hash of `case => reducers functions`, it's only a suggestion and maybe too big of a change for this case!

File: mailpoet/assets/js/src/analytics.js (hunk 1 above)
Comment (lysyjan) [reply mailpoet/assets/js/src/analytics.js]: Thanks for this catch!

**Comment by lysyjan** [comment]
@triple0t Thank you for your suggestions. I added two commits with changes. I guess the readability is better now, but I want to avoid a robust solution that mentioned @samnajian.

**Review by triple0t — APPROVED** [review]
Looks good 💪🏿

**Review by Aschepikov — APPROVED** [review]
This is verified, good job

**Pull request merged**